ethereum/contracts/bridge/L1ERC20Bridge.sol

```diff
@@ -77,31 +91,56 @@ contract L1ERC20Bridge is IL1Bridge, IL1BridgeLegacy, AllowListed, ReentrancyGua
     /// @notice At the time of the function call, it is not yet deployed in L2, but knowledge of its address
     /// @notice is necessary for determining L2 token address by L1 address, see `l2TokenAddress(address)` function
     /// @param _governor Address which can change L2 token implementation and upgrade the bridge
-    /// @param _deployBridgeImplementationFee How much of the sent value should be allocated to deploying the L2 bridge
     /// implementation
-    /// @param _deployBridgeProxyFee How much of the sent value should be allocated to deploying the L2 bridge proxy
     function initialize(
         bytes[] calldata _factoryDeps,
         address _l2TokenBeacon,
-        address _governor,
-        uint256 _deployBridgeImplementationFee,
-        uint256 _deployBridgeProxyFee
+        address _l2Bridge,
+        address _governor
     ) external payable reentrancyGuardInitializer {
         require(_l2TokenBeacon != address(0), "nf");
         require(_governor != address(0), "nh");
         // We are expecting to see the exact three bytecodes that are needed to initialize the bridge
         require(_factoryDeps.length == 3, "mk");
         // The caller miscalculated deploy transactions fees
-        require(msg.value == _deployBridgeImplementationFee + _deployBridgeProxyFee, "fee");
         l2TokenProxyBytecodeHash = L2ContractHelper.hashL2Bytecode(_factoryDeps[2]);
         l2TokenBeacon = _l2TokenBeacon;
+        l2Bridge = _l2Bridge;
```

Comment:
Oh, ok, this will be a bit of a pain, since, WETH on ERA is already deployed, using a different method. For that I will have to track that it is on a different address.

Reply:
I now track each bridge Address on each L2. We will still have a standard bridge Bytecode, which I will include with the deployment scripts.